### PR TITLE
Project 19: Happy Thoughts API

### PR DESCRIPTION
**Pull request by Thessan**
[https://thessans-happythoughts-api.herokuapp.com/](https://thessans-happythoughts-api.herokuapp.com/)

https://send-a-happy-thought.netlify.app

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -7,23 +7,78 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
...
 app.use(bodyParser.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+
+// ROUTES
+
+app.get('/', (request, response) => {
+  response.send('Welcome to my Happy Thoughts API 💌')
```
Comment (BamStugendal) [inline server.js:40]: Nice welcome message 😄

File: server.js [hunk 2]
```diff
@@ -7,23 +7,78 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
...
+})
+
+// GET the latest 20 thoughts
+app.get('/thoughts', async (request, response) => {
```
Comment (BamStugendal) [inline server.js:44]: Interesting that you use whole words here, makes it easier to understands want it really is/does 😊

File: server.js [hunk 3]
```diff
@@ -7,23 +7,78 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
...
+    catch (err) {
+    response.status(400).json({ message: 'Sorry, could not save thought to the database', errors: err.errors
+    })
+  }
+})
+
+// POST a like to a specific thought
+app.post('/thoughts/:thoughtId/like', async (request, response) => {
+  try {
+    await Thought.updateOne({ _id: request.params.thoughtId }, { $inc: { hearts: 1 } })
```
Comment (BamStugendal) [inline server.js:72]: Good that used `updateOne` instead of `update` 👍

**Review by BamStugendal — APPROVED** [review]
Great job Therése 👍  Clean code and nice commenting too 😄

File: README.md [hunk 4]
```diff
@@ -10,4 +10,4 @@ Describe how you approached to problem, and what tools and techniques you used t
 
 ## View it live
 
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+https://thessans-happythoughts-api.herokuapp.com/
```
Comment (KarolCislowski) [inline README.md:13]: It is very important to make your README file outstanding 😺 Most Recruiters will assess your work using it and you need to make them want to pass this repo to developers 😸

**Review by KarolCislowski — APPROVED** [review]
This is a very nice and clean project 🎆 
Remember to revisit the Readme and give it some love 😺

**Pull request closed without merging**